### DISCUSSION TITLE
Reverse proxy for plausible, written in PHP for Apache

### DISCUSSION THREAD

**Discussion by Alpine418** [plausible/analytics#1088] [Anything Else]
Hi folks,

I'm using a shared hosting and dont have access to a native reverse proxy like mod_proxy in Apache. So I wrote a small reverse proxy in PHP.

Please check the gist here:
https://gist.github.com/rjgamer/9bdae878b2ceddc3ef10e3c4d61f9326

It looks like it works for me. Anybody interested in a few tests or any feedback?

Thanks,
rjgamer

**Comment by Alpine418** [comment]
Hello there?

**Reply by ukutaht** [reply]
Hey, thanks for contributing this. I don't run any PHP myself, hopefully someone from the community is interested in testing this one out.

**Reply by gitlabisbetterthangithub** [reply]
I'd be happy to test it out

**Reply by gitlabisbetterthangithub** [reply]
Works fine for me

**Reply by Alpine418** [reply]
Thanks for testing. 👍

**Reply by metmarkosaric** [reply]
thank you both! we'll see to include these in the docs!

**Reply by Alpine418** [reply]
Found an issue. The geolocation is always used from the proxy instead of the client.

How does plausible read the geolocation? Can I send it as an header?

**Reply by Alpine418** [reply]
https://stackoverflow.com/a/1301417

@ukutaht May this is the solution as headers instead of spoofing. Would this work?

**Reply by ukutaht** [reply]
We use the `X-Forwarded-For` header to get the client IP address. I think the linked code would work although the `REMOTE_ADDR` header is unnecessary, just `X-Forwarded-For` will suffice.

**Reply by Alpine418** [reply]
Perfect!

Updated the gist: https://gist.github.com/rjgamer/9bdae878b2ceddc3ef10e3c4d61f9326

**Comment by Alpine418** [comment]
Fixed missing support for geolocation by adding X-Forwarded-For header.

https://gist.github.com/rjgamer/9bdae878b2ceddc3ef10e3c4d61f9326

Will move the gist to a github repository.

**Reply by Alpine418** [reply]
Repository online:
https://github.com/Neoflow/ReverseProxy-PlausibleAnalytics

**Reply by ukutaht** [reply]
Great, thanks @rjgamer !
